### DISCUSSION TITLE
How to add new Clusters to an End Device

### DISCUSSION THREAD

**Discussion by cw2802** [zigpy/zha-device-handlers#2880] [Q&A] [answered]
Hello, i´m fairly new to Home Assistant and Home Automation in general. I´m trying to achieve the following:

I have a TuyaSmartRemote0042TO Button and a Curtain Switch TS130F. I´d like to be able to bind these two together, so that whenever I click on the button on my Remote the Curtain Switch should either go up or down (Without using a configured Automation in HA).

I think I understood the basic concept of the quirks. What beats me at the moment is, how can I add the "TuyaCoveringCluster" to the Remote and make it send commands to the very same cluster that my curtain switch is using.

I thought I just have to add the corresponding class (I simply copy/pasted the TuyaCoveringCluster from the Curtain Switch quirk) to the quirk (I´m using the [ts0042 quirk](https://github.com/zigpy/zha-device-handlers/blob/dev/zhaquirks/tuya/ts0042.py) from this repo) and use the device automation triggers to send the command to the accoding cluster. But that didn´t had any effect.

Do I also have to add the Command to my Copied TuyaCoveringCluster class or am I missing something else? Any help/hint would be appreciated.

Kind regards,
Chris

**Comment by MattWestb** [comment, marked as answer]
Then both of your device is only using zigbee as one transport but not using zigbee standard commands and is not certified they cant being bond as normal devices can being dome (like IKEA blinds and there remotes) and is working also then the host system is not working On / offline.
You can using the events the remote is sending (also device automatons from the remote device card) and trigger open and closing commands to sending to the curtain switch but its on online things and can having delays or not working at all if the network and / or host system have problems.

**Reply by cw2802** [reply]
Thanks for the answer. Now that I think about it, it makes perfect sense, as the quirk is only changing the behavior within HA but not on the device itself.

**Comment by Hedda** [comment]
> hen both of your device is only using zigbee as one transport but not using zigbee standard commands and is not certified they cant being bond as normal devices can being dome (like IKEA blinds and there remotes) and is working also then the host system is not working On / offline.
> You can using the events the remote is sending (also device automatons from the remote device card) and trigger open and closing commands to sending to the curtain switch but its on online things and can having delays or not working at all if the network and / or host system have problems.

@MattWestb could you maybe post that explanation in ZHA docs for reference?

https://www.home-assistant.io/integrations/zha

perhaps as a new section under the troubleshooting?

https://www.home-assistant.io/integrations/zha#troubleshooting

maybe as a sub-section under "knowing which devices are supported"

https://www.home-assistant.io/integrations/zha#knowing-which-devices-are-supported
